airbyte-integrations/connectors/destination-oracle/src/test-integration/java/io/airbyte/integrations/destination/oracle/OracleIntegrationTest.java

```diff
@@ -0,0 +1,194 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.oracle;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.db.Database;
+import io.airbyte.db.Databases;
+import io.airbyte.integrations.base.JavaBaseConstants;
+import io.airbyte.integrations.destination.ExtendedNameTransformer;
+import io.airbyte.integrations.standardtest.destination.DestinationAcceptanceTest;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.jooq.JSONFormat;
+import org.jooq.JSONFormat.RecordFormat;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.testcontainers.containers.OracleContainer;
+
+public class OracleIntegrationTest extends DestinationAcceptanceTest {
+
+  private static final JSONFormat JSON_FORMAT = new JSONFormat().recordFormat(RecordFormat.OBJECT);
+
+  private static OracleContainer db;
+  private ExtendedNameTransformer namingResolver = new OracleNameTransformer();
+  private JsonNode configWithoutDbName;
+  private JsonNode config;
+
+  @BeforeAll
+  protected static void init() {
+    db = new OracleContainer("epiclabs/docker-oracle-xe-11g");
+    db.start();
+  }
+
+  @Override
+  protected String getImageName() {
+    return "airbyte/destination-oracle:dev";
+  }
+
+  private JsonNode getConfig(OracleContainer db) {
+    return Jsons.jsonNode(ImmutableMap.builder()
+        .put("host", db.getHost())
+        .put("port", db.getFirstMappedPort())
+        .put("username", db.getUsername())
+        .put("password", db.getPassword())
+        .put("schema", "testSchema")
+        .put("sid", db.getSid())
+        .build());
+  }
+
+  @Override
+  protected JsonNode getConfig() {
+    return config;
+  }
+
+  @Override
+  protected JsonNode getFailCheckConfig() {
+    return Jsons.jsonNode(ImmutableMap.builder()
+        .put("host", db.getHost())
+        .put("username", db.getUsername())
+        .put("password", "wrong password")
+        .put("schema", "public")
+        .put("port", db.getFirstMappedPort())
+        .put("sid", db.getSid())
+        .put("ssl", false)
+        .build());
+  }
+
+  @Override
+  protected List<JsonNode> retrieveRecords(TestDestinationEnv env, String streamName, String namespace) throws Exception {
+    return retrieveRecordsFromTable(namingResolver.getRawTableName(streamName), namespace)
+        .stream()
+        .map(r -> Jsons.deserialize(r.get(JavaBaseConstants.COLUMN_NAME_DATA.substring(1).toUpperCase()).asText()))
```

Comment:
```
JavaBaseConstants.COLUMN_NAME_DATA.substring(1).toUpperCase()
```
what's this? would it make sense to put this as a contsant in `OracleDestination` so we have record of what the actual column names are? maybe do the same for ID and EMITTED_AT as well?

Reply:
is there a reason why you're doing this transformation out here instead of from within `retrieveRecordsFromTable`?

Reply:
> what's this?

As mentioned above, `substring(1)` is necessary to strip off the leading underscore.  The `toUpperCase` is there because Oracle converts all identifiers to upper case, and the record set identifier lookup is case-sensitive.

Reply:
yeah. makes sense.
1. can we move this line into `retrieveRecordsFromTable`? i'm not sure why it is separate out from the rest of that logic.
2. can you declare the oracle versions of the column names in the `OracleDestination` class as static field? That way we very clearly say what the expect column names are instead of just doing the conversion in an ad hoc way.

Reply:
1. No, because `retrieveRecordsFromTable` is also called elsewhere.
2. Sure, that would be simple enough.